README.rst

```diff
@@ -0,0 +1,329 @@
+`CDAP application <http://docs.cdap.io/cdap/current/en/developers-manual/building-blocks/applications.html>`__
+that uses ingested raw purchase events (of the form '``<name> bought <n> <item>s for $<price>``') to compute
+the total purchases made by a user along with the total purchases made for a particular product.
+
+You will:
+
+- Build ``PurchaseEventParser``, a
+  `MapReduce program <http://docs.cdap.io/cdap/current/en/developers-manual/building-blocks/mapreduce-programs.html>`__
+  to parse the raw purchase events and create ``Purchase`` objects from them;
+- Build ``PurchaseCounterByUser``, a
+  `MapReduce program <http://docs.cdap.io/cdap/current/en/developers-manual/building-blocks/mapreduce-programs.html>`__
+  to count the purchases made per user;
+- Build ``PurchaseCounterByProduct``, a
+  `MapReduce program <http://docs.cdap.io/cdap/current/en/developers-manual/building-blocks/mapreduce-programs.html>`__
+  to count the purchases made per product;
+- Build ``PurchaseWorkflow``, a
+  `Workflow program <http://docs.cdap.io/cdap/current/en/developers-manual/building-blocks/workflows.html>`__
+  which will first execute the MapReduce program ``PurchaseEventParser``. If the predicate ``PurchaseEventVerifier``
+  evaluates to true, the workflow will in parallel execute the MapReduce program ``PurchaseCounterByUser`` and
+  ``PurchaseCounterByProduct`` otherwise, it will execute the action ``EmailNotifier``;
+- Use a
+  `Dataset <http://docs.cdap.io/cdap/current/en/developers-manual/building-blocks/datasets/index.html>`__
+  to persist results of the MapReduce programs; and
+- Build a
+  `Service <http://docs.cdap.io/cdap/current/en/developers-manual/building-blocks/services.html>`__
+  to serve the results via HTTP.
+
+What You Will Need
+==================
+
+- `JDK 7 <http://www.oracle.com/technetwork/java/javase/downloads/index.html>`__
+- `Apache Maven 3.1+ <http://maven.apache.org/>`__
+- `CDAP SDK <http://docs.cdap.io/cdap/current/en/developers-manual/getting-started/standalone/index.html>`__
+
+Let’s Build It!
+===============
+
+The following sections will guide you through building an application from scratch. If you
+are interested in deploying and running the application right away, you can clone its
+source code from this GitHub repository. In that case, feel free to skip the next two
+sections and jump right to the
+`Build and Run Application <#build-and-run-application>`__ section.
+
+Application Design
+------------------
+
+The application will assume that the purchase events are ingested
+into a Stream. The events can be ingested into a Stream continuously
+in real time or in batches; whichever way, it doesn’t affect the ability
+of the MapReduce programs to consume them.
+
+The ``PurchaseWorkflow`` encapsulates the set of MapReduce programs, which extracts the required information from the
+raw purchase events and computes the total purchases made by a particular user and total purchases made for a
```

Comment:
use "each" instead of "particular" to emphasize that it does so for all users/products, not for a particular one?